sql/executor.go

```diff
@@ -955,34 +955,31 @@ func (e *Executor) execStmtInOpenTxn(
 		return Result{PGTag: s.StatementTag()}, nil
 	}
 
-	if txnState.tr != nil {
-		txnState.tr.LazyLog(stmt, true /* sensitive */)
-	}
+	session := planMaker.session
+	log.Event(session.context, stmt.String())
 
 	result, err := e.execStmt(stmt, planMaker, implicitTxn /* autoCommit */)
 	if err != nil {
 		if traceSQL {
-			log.Tracef(txnState.txn.Context, "ERROR: %v", err)
-		}
-		if txnState.tr != nil {
-			txnState.tr.LazyPrintf("ERROR: %v", err)
+			log.ErrEventf(txnState.txn.Context, "ERROR: %v", err)
 		}
+		log.ErrEventf(session.context, "ERROR: %v", err)
```

Comment:
The statements in the `switch` above mostly early-return, so they won't ever hit the events calls. That'll look confusing in the event logs.
Don't make any changes here, but perhaps this would make a good follow-up.